superset/sql_parse.py

```diff
@@ -287,9 +292,15 @@ def _extract_tables_from_sql(self) -> set[Table]:
         """
         try:
             statements = parse(self.sql, dialect=self._dialect)
-        except ParseError:
-            logger.warning("Unable to parse SQL (%s): %s", self._dialect, self.sql)
-            return set()
+        except ParseError as ex:
+            dialect = self._dialect or "generic"
+            raise SupersetSecurityException(
+                SupersetError(
+                    error_type=SupersetErrorType.QUERY_SECURITY_ACCESS_ERROR,
+                    message=__(f"Unable to parse SQL ({dialect}): {self.sql}"),
+                    level=ErrorLevel.ERROR,
```

Comment:
@betodealmeida is this a user error or system error?

Reply:
It's hard to tell. It could be invalid SQL (user error), or it could be because limitations of the parser (system error).

Reply:
Ok, that makes sense. I'm thinking of a scenario where someone is making a lot invalid sql calls and it just keeps loading up a system monitoring queue with what looks to be an outage of a sort.

Reply:
Yeah, I'm worried about that too, and I think that's why originally the behavior was to not to fail. But the alternative is allowing people to run queries on tables they don't have access because we fail to extract the reference from the SQL.

Reply:
@betodealmeida I _think_ having Superset being less lenient/blasé is likely a good thing, i.e., previously it blindly would could have* given a user access whenever it wasn't able to parse the SQL, irrespective of whether the user had access to the underly data.

The major challenge I see is if SQLGlot wrongfully isn't able to parse a SQL statement and thus we're wrongfully blocking a query. How do we help unblock users here?

\* This is dependent on how one configured their security manager.